scrapy/core/downloader/handlers/http11.py

```diff
@@ -31,8 +32,16 @@ def __init__(self, settings):
         self._pool = HTTPConnectionPool(reactor, persistent=True)
         self._pool.maxPersistentPerHost = settings.getint('CONCURRENT_REQUESTS_PER_DOMAIN')
         self._pool._factory.noisy = False
+
+        self._sslMethod = openssl_methods[settings.get('DOWNLOADER_CLIENT_TLS_METHOD')]
         self._contextFactoryClass = load_object(settings['DOWNLOADER_CLIENTCONTEXTFACTORY'])
-        self._contextFactory = self._contextFactoryClass()
+        # try method-aware context factory
+        try:
+            self._contextFactory = self._contextFactoryClass(method=self._sslMethod)
+        except TypeError:
+            # use defaults
+            self._contextFactory = self._contextFactoryClass()
```

Comment:
thx @dangra .
It's a bit nicer now in https://github.com/redapple/scrapy/commit/0336c2504af45a87d2aae4c1ed45039c0e0ea183 :

```
2016-02-24 16:41:46 [py.warnings] WARNING: /home/paul/src/scrapy/scrapy/core/downloader/handlers/http11.py:50: UserWarning: 
 'sslissues.contextfactory.TLSFlexibleContextFactory' does not accept `method` argument (type OpenSSL.SSL method, e.g. OpenSSL.SSL.SSLv23_METHOD). Please upgrade your context factory class to handle it or ignore it.
  warnings.warn(msg)

```